pkg/nsx/services/subnet/builder.go

```diff
@@ -0,0 +1,109 @@
+package subnet
+
+import (
+	"fmt"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+var (
+	String = common.String
+	Int64  = common.Int64
+	Bool   = common.Bool
+)
+
+func getCluster(service *SubnetService) string {
+	return service.NSXConfig.Cluster
+}
+
+func (service *SubnetService) buildSubnet(obj client.Object, tags []model.Tag) (*model.VpcSubnet, error) {
+	tags = append(tags, service.buildBasicTags(obj)...)
+	var nsxSubnet *model.VpcSubnet
+	switch o := obj.(type) {
+	case *v1alpha1.Subnet:
+		nsxSubnet = &model.VpcSubnet{
+			Id:             String(string(o.GetUID())),
+			AccessMode:     String(string(o.Spec.AccessMode)),
+			DhcpConfig:     service.buildDHCPConfig(&o.Spec.DHCPConfig),
+			Tags:           tags,
+			AdvancedConfig: service.buildAdvancedConfig(&o.Spec.AdvancedConfig),
+		}
+	case *v1alpha1.SubnetSet:
+		index := len(o.Status.Subnets)
+		nsxSubnet = &model.VpcSubnet{
+			Id:             String(fmt.Sprintf("%s-%d", string(o.GetUID()), index)),
+			AccessMode:     String(string(o.Spec.AccessMode)),
+			DhcpConfig:     service.buildDHCPConfig(&o.Spec.DHCPConfig),
+			Tags:           tags,
+			AdvancedConfig: service.buildAdvancedConfig(&o.Spec.AdvancedConfig),
+		}
+	default:
+		return nil, SubnetTypeError
+	}
+	return nsxSubnet, nil
+}
+
+func (service *SubnetService) buildDHCPConfig(obj *v1alpha1.DHCPConfig) *model.DhcpConfig {
+	// Subnet DHCP is used by AVI, not needed for now. We need to explicitly mark enableDhcp = false,
+	// otherwise Subnet will use DhcpConfig inherited from VPC.
+	dhcpConfig := &model.DhcpConfig{
+		//DhcpRelayConfigPath: String(obj.DHCPRelayConfigPath),
+		//DhcpV4PoolSize:      Int64(int64(obj.DHCPV4PoolSize)),
+		//DhcpV6PoolSize:      Int64(int64(obj.DHCPV6PoolSize)),
+		//DnsClientConfig:     service.buildDNSClientConfig(&obj.DNSClientConfig),
+		EnableDhcp: Bool(false),
```

Comment:
when enableDHCP: false, will only static ip pool created for the subnet?

Reply:
yes